srv/fluffi/data/fluffiweb/app/controllers.py

```diff
@@ -1223,14 +1212,44 @@ def insertFormInputForProject(form, request):
         engine.dispose()
 
 
-def addTargetZipToFuzzjob(projId, targetFileName):
-    deployment_package = models.DeploymentPackages(name=targetFileName)
-    db.session.add(deployment_package)
-    db.session.commit()
-    fuzzjob_deployment_package = models.FuzzjobDeploymentPackages(Fuzzjob=projId,
-                                                                  DeploymentPackage=deployment_package.ID)
-    db.session.add(fuzzjob_deployment_package)
-    db.session.commit()
+def uploadNewTargetZipHandler(projId, targetFile):
+    try:
+        targetFileData = targetFile.read()
+        targetFileName = targetFile.filename
+    except Exception as e:
+        print(e)
+        return "Error: Failed to read file!", "error"
+    
+    try:
+        existingPackage = db.session.query(models.DeploymentPackages, models.FuzzjobDeploymentPackages).filter(
+                models.DeploymentPackages.ID == models.FuzzjobDeploymentPackages.DeploymentPackage).filter(
+                    models.FuzzjobDeploymentPackages.Fuzzjob == projId).first()                                            
+    except Exception as e:
+        print(e)
+        return "Error: Failed to query existing packages!", "error"
+    
+    if existingPackage is not None:  
+        try:
+            FTP_CONNECTOR.saveTargetFileOnFTPServer(targetFileData, existingPackage.DeploymentPackages.name)
+            return "Success: Updated target!", "success"
+        except Exception as e:
+            print(e)  
+            return "Error: Failed to update target!", "error"                                                     
+    try:
+        FTP_CONNECTOR.saveTargetFileOnFTPServer(targetFileData, targetFileName)
+        
+        newDeploymentPackage = models.DeploymentPackages(name=targetFileName)
+        db.session.add(newDeploymentPackage)
+        db.session.commit()
+        
+        newFjDeploymentPackage = models.FuzzjobDeploymentPackages(Fuzzjob=projId, DeploymentPackage=newDeploymentPackage.ID)
+        db.session.add(newFjDeploymentPackage)
+        db.session.commit()
+        
+        return "Success: Uploaded new target!", "success"
```

Comment:
![image](https://user-images.githubusercontent.com/17053360/88145993-8feb1700-cbfb-11ea-84db-18d9feaf65dc.png)

Please remove one "Success".

Reply:
This seems to be the case several times.